From 3dfb0e9e8a4369705e1ac2e5603ad3068be61061 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 24 Apr 2023 10:26:07 +0200
Subject: [PATCH 0301/1695] dts: stm32: populate RISAF nodes

Populate RISAF1/2/4/5 nodes. Default status for RISAF4 is okay as it
protects the DDR memory region and RISAF1 that protects BKPSRAM.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 441552af6cd0759ad0389951eebc15bb55e8375b dts: stm32: populate RISAF nodes]
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/326327
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: Ieda0af69c30ba1339ebcb6b121c269a7fb87dbbb
---
 core/arch/arm/dts/stm32mp251.dtsi | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 0e0518729..933c833d1 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -227,22 +227,32 @@
 		};
 
 		risaf1: risaf@420a0000 {
+			compatible = "st,stm32mp25-risaf";
 			reg = <0x420a0000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_BUS_BKPSRAM>;
+			st,mem-map = <0x0 0x42000000 0x0 0x2000>;
 		};
 
 		risaf2: risaf@420b0000 {
+			compatible = "st,stm32mp25-risaf";
 			reg = <0x420b0000 0x1000>;
+			clocks = <&rcc CK_KER_OSPI1>;
+			st,mem-map = <0x0 0x60000000 0x0 0x10000000>;
 			status = "disabled";
 		};
 
 		risaf4: risaf@420d0000 {
+			compatible = "st,stm32mp25-risaf-enc";
 			reg = <0x420d0000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_BUS_RISAF4>;
+			st,mem-map = <0x0 0x80000000 0x1 0x00000000>;
 		};
 
 		risaf5: risaf@420e0000 {
+			compatible = "st,stm32mp25-risaf";
 			reg = <0x420e0000 0x1000>;
+			clocks = <&rcc CK_BUS_PCIE>;
+			st,mem-map = <0x0 0x10000000 0x0 0x10000000>;
 			status = "disabled";
 		};
 
-- 
2.47.2

